### DISCUSSION TITLE
Contributing to Examples

### DISCUSSION THREAD

**Discussion by luke-h1** [urql-graphql/urql#1842] [Q&A] [answered]
Hi, I'm just wondering if it's acceptable for me to make a PR to add an example of TypeGraphQL, Next.js & Formik image uploads to the examples folder? This is something that puzzled me for quite a while so just thought it'd be a nice addition to the examples in this repo for anyone else facing the same issues

**Comment by JoviDeCroock** [comment, marked as answer]
Currently our examples are quite atomic little front-end projects that highlight a particular use-case like in this case, `FileUpload` here will highlight the file-upload one. https://github.com/FormidableLabs/urql/tree/main/examples/with-multipart/src I'd imagine this being highly similar to using any form library. We've also attempted not to capture any specific back-end in these examples as that might cause people to think that we're focussing on specific graphql-server technologies 😅 

The difference with the aforementioned example might be that instead of submitting directly you'll store it and wait for `onSubmit` to pass it to your mutation. If I'm wrong here I'd be happy to hear any complications you've been facing here 😅

**Reply by luke-h1** [reply]
Ah right ok cool, that all sounds good to me! Yeah, I had to wait for the onSubmit to pass along the file with the main mutation. I think the main thing that made it hard was this was my first component where I needed to upload files to a server so it was a little tricky to figure out the quirks of Formik 😅.Nothing wrong with the docs imo. My solution was: 
``` 
<input
                type="file"
                accept="image/*"
                onChange={({ target: { validity, files } }) => {
                  if (validity.valid && files) {
                    setFieldValue('image', files[0]);
                    // set 'file' of the form data as files[0]
                  }
                }}
              />
```
And then in `onSubmit` I passed `values.image` to my mutation

**Reply by JoviDeCroock** [reply]
Yeah that looks similar to our example and what I had in mind 😅I'm not sure if Formik in general has examples on this (in casey ou had weird quirks that might be an appropriate place) 😅
